compiler/semantic/sql.go

```diff
@@ -0,0 +1,457 @@
+			seq = append(seq, badOp())
+		} else {
+			seq = append(seq, groupby)
+			if sel.Having != nil {
+				seq = append(seq, dag.NewFilter(a.semExpr(sel.Having)))
+			}
+		}
+	} else if sel.Args != nil {
+		if sel.Having != nil {
+			a.error(sel, errors.New("HAVING clause used without GROUP BY"))
+			seq = append(seq, badOp())
+		}
+		// GroupBy will do the cutting but if there's no GroupBy,
+		// then we need a cut for the select expressions.
+		// For SELECT *, cutter is nil.
+		selector, err := convertSQLSelect(selection)
+		if err != nil {
+			a.error(sel, err)
+			seq = append(seq, badOp())
+		} else {
+			seq = append(seq, selector)
+		}
+	}
+	if len(seq) == 0 {
+		seq = []dag.Op{dag.PassOp}
+	}
+	return seq
+}
+
+func (a *analyzer) semSelectValue(sel *ast.Select, seq dag.Seq) dag.Seq {
+	if sel.GroupBy != nil {
+		a.error(sel, errors.New("SELECT VALUE cannot be used with GROUP BY"))
+		seq = append(seq, badOp())
+	}
+	if sel.Having != nil {
+		a.error(sel, errors.New("SELECT VALUE cannot be used with HAVING"))
+		seq = append(seq, badOp())
+	}
+	exprs := make([]dag.Expr, 0, len(sel.Args))
+	for _, assignment := range sel.Args {
+		if assignment.LHS != nil {
+			a.error(sel, errors.New("SELECT VALUE cannot AS clause in selection"))
+		}
+		exprs = append(exprs, a.semExpr(assignment.RHS))
+	}
+	seq = append(seq, &dag.Yield{
+		Kind:  "Yield",
+		Exprs: exprs,
+	})
+	if sel.Where != nil {
+		seq = append(seq, dag.NewFilter(a.semExpr(sel.Where)))
+	}
+	return seq
+}
+
+func (a *analyzer) semSQLOp(op ast.Op, seq dag.Seq) dag.Seq {
+	switch op := op.(type) {
+	case *ast.SQLPipe:
+		if len(seq) > 0 {
+			panic("semSQLOp: SQL pipes can't have parents")
+		}
+		return a.semSeq(op.Ops)
+	case *ast.Select:
+		return a.semSelect(op, seq)
+	case *ast.SQLJoin:
+		return a.semSQLJoin(op, seq)
+	case *ast.OrderBy:
+		nullsFirst, ok := nullsFirst(op.Exprs)
+		if !ok {
+			a.error(op, errors.New("differring nulls first/last clauses not yet supported"))
+			return append(seq, badOp())
+		}
+		var exprs []dag.SortExpr
+		for _, e := range op.Exprs {
+			exprs = append(exprs, a.semSortExpr(e))
+		}
+		return append(a.semSQLOp(op.Op, seq), &dag.Sort{
+			Kind:       "Sort",
+			Args:       exprs,
+			NullsFirst: nullsFirst,
+			Reverse:    false, //XXX this should go away
+		})
+	case *ast.Limit:
+		e := a.semExpr(op.Count)
+		var err error
+		val, err := kernel.EvalAtCompileTime(a.zctx, e)
+		if err != nil {
+			a.error(op.Count, err)
+			return append(seq, badOp())
+		}
+		if !super.IsInteger(val.Type().ID()) {
+			a.error(op.Count, fmt.Errorf("expression value must be an integer value: %s", zson.FormatValue(val)))
+			return append(seq, badOp())
+		}
+		limit := val.AsInt()
+		if limit < 1 {
+			a.error(op.Count, errors.New("expression value must be a positive integer"))
+		}
+		head := &dag.Head{
+			Kind:  "Head",
+			Count: int(limit),
+		}
+		return append(a.semSQLOp(op.Op, seq), head)
+	default:
+		panic(fmt.Sprintf("semSQLOp: unknown op: %T", op))
```

Comment:
Nit: More detail via %#v in all these switch-default panics could be nice.
```suggestion
		panic(fmt.Sprintf("semSQLOp: unknown op: %#v", op))
```